### PR TITLE
Add more test to verify the correctness of `multi-batch-write`

### PR DESCRIPTION
**Pull request by Little-Wallace**
I write more test to verify the correctness of `multi-batch-write`. I try to add `SyncPoint` in every step of `MultiBatchWrite` to verify it. It may be a little incomprehensible.

### CODE REVIEW

File: db/db_write_test.cc [hunk 1]
```diff
@@ -185,66 +185,6 @@ TEST_P(DBWriteTest, LockWalInEffect) {
   ASSERT_OK(dbfull()->UnlockWAL());
 }
 
-TEST_P(DBWriteTest, MultiThreadWrite) {
```
Comment (yiwu-arbug) [inline db/db_write_test.cc]: Can we keep the test? If the test is broken with multi-batch write,  simply disable it for multi-batch write.

Comment (Little-Wallace) [reply db/db_write_test.cc:188]: I did not remove it. I just moved it into `db/db_multi_batch_write_test.cc`.

**Comment by yiwu-arbug** [comment]
Closing old PRs with no progress.

**Pull request closed without merging**